libcaf_net/src/ip.cpp

```diff
@@ -0,0 +1,136 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "caf/config.hpp"
+#include "caf/detail/socket_sys_includes.hpp"
+#include "caf/error.hpp"
+#include "caf/ip_address.hpp"
+#include "caf/logger.hpp"
+#include "caf/string_view.hpp"
+
+// clang-format off
+#ifdef CAF_WINDOWS
+#  ifndef _WIN32_WINNT
+#    define _WIN32_WINNT 0x0600
+#  endif
+#  include <iphlpapi.h>
+#else
+#  include <net/if.h>
+#  include <netdb.h>
+#  include <ifaddrs.h>
+#  include <sys/ioctl.h>
+#  include <sys/types.h>
+#endif
+// clang-format on
+
+#ifndef HOST_NAME_MAX
+#  define HOST_NAME_MAX 255
+#endif
+
+using std::pair;
+using std::string;
+using std::vector;
+
+namespace caf {
+namespace net {
+namespace ip {
+
+namespace {
+
+void* fetch_in_addr(int family, sockaddr* addr) {
+  if (family == AF_INET)
+    return &reinterpret_cast<sockaddr_in*>(addr)->sin_addr;
+  return &reinterpret_cast<sockaddr_in6*>(addr)->sin6_addr;
+}
+
+// TODO: Use getnameinfo instead?
+int fetch_addr_str(bool get_ipv4, bool get_ipv6, char (&buf)[INET6_ADDRSTRLEN],
```

Comment:
The flags are unnecessary: The only function that ever calls `fetch_addr_str` always sets both to `true`.

```suggestion
int fetch_addr_str(char (&buf)[INET6_ADDRSTRLEN],
```